activerecord/test/models/author.rb

```diff
@@ -4,6 +4,7 @@ class Author < ActiveRecord::Base
   has_one :post
   has_many :very_special_comments, :through => :posts
   has_many :posts_with_comments, -> { includes(:comments) }, :class_name => "Post"
+  has_many :posts_ordered_by_comments_tags_count, -> { joins('LEFT JOIN comments ON comments.post_id = posts.id').order("SUM(comments.tags_count)").group('posts.id') }, :class_name => "Post"
```

Comment:
Instead of adding the association globally, can you create a new subclass in the test and add it there?

```ruby
subclass = Class.new(Author) do
  has_many :posts_ordered_by_comments_tags_count, -> { joins('LEFT JOIN comments ON comments.post_id = posts.id').order("SUM(comments.tags_count)").group('posts.id') }, :class_name => "Post"
end
```